tests/auto/desktop-qt5/useragent/tst_useragent.qml

```diff
@@ -12,12 +13,34 @@ Item {
     readonly property string override_developer: "Developer specified override"
     readonly property string test_page: "https://browser.sailfishos.org/tests/testuseragent.html"
 
+    Connections {
+        target: QmlMozContext
+        onOnInitialized: {
+            QmlMozContext.addComponentManifest(TestHelper.getenv("QTTESTSROOT") + "/components/TestHelpers.manifest")
+        }
+    }
+
     QmlMozView {
         id: webViewport
         visible: true
         focus: true
         active: true
         anchors.fill: parent
+        property string useragent: ""
+
+        onViewInitialized: {
+            webViewport.loadFrameScript("chrome://tests/content/testHelper.js")
+            webViewport.addMessageListener("testembed:useragent")
+        }
```

Comment:
No need to handle these on viewInitialized signal handler rather Component.onCompleted is enough.